src/unix/fs.c

```diff
@@ -129,8 +129,27 @@
 static ssize_t uv__fs_fdatasync(uv_fs_t* req) {
 #if defined(__linux__) || defined(__sun) || defined(__NetBSD__)
   return fdatasync(req->file);
-#elif defined(__APPLE__) && defined(SYS_fdatasync)
-  return syscall(SYS_fdatasync, req->file);
+#elif defined(__APPLE__) && defined(F_FULLFSYNC)
+  /* Apple's fdatasync explicitly does NOT flush the drive write cache to the
+   * drive platters. This is in contrast to Linux's fdatasync which does,
+   * according to recent man pages. F_FULLFSYNC is Apple's equivalent for
+   * flushing buffered data to permanent storage.
+   */
+  return fcntl(req->file, F_FULLFSYNC);
+#else
+  return fsync(req->file);
```

Comment:
I thought hard about doing that. If we did that, we'd still need a comment explaining why we're delegating to `uv__fs_fsync()` rather than using Apple's fdatasync syscall, otherwise someone might replace the alias with fdatasync itself. So I did like it is, also avoiding the `uv__fs_fsync()` dependency across functions.